src/main/java/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -199,7 +196,6 @@ protected void purchase(final boolean purchaseForBid, final int pusToSpend, fina
       } finally {
         data.releaseReadLock();
       }
-      calc.setData(dataCopy);
```

Comment:
@ron-murhammer Kind of related to #2600...  In the current implementation, it doesn't look like `dataCopy` will be used by the Battle Calculator anymore.  Even though it's passed to `simulateBattles()`, it never gets passed to the calculator, which ends up just using the original game data.

Just want to confirm if not using this possibly-different copy of the `GameData` is a problem that needs to be fixed, as well.

Reply:
The AI needs to pass a copy of game data when it simulates move and battles before purchasing. So either needs to set a copy of the data or create a new battle calc with the data. This is definitely not ok to have removed.

Reply:
Ok, I'll see how difficult that is to work into the new architecture.  A complete revert is looking more attractive, though. 😞

Reply:
@ssoloff Yeah. I'm fine with a full revert on this if that is the easiest path forward.